### PR TITLE
Add Integration Index to Demo

### PR DESCRIPTION
**Pull request by Swiddis**
### Description
Alternative approach to #1520 
TODO: Testing

### Issues Resolved
[List any issues this PR will resolve]

### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass, including unit test, integration test and doctest
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
  - [ ] New functionality has user manual doc added
- [ ] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: src/main/kotlin/org/opensearch/integrations/index/IntegrationIndex.kt [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+import org.opensearch.integrations.model.Integration
+import org.opensearch.observability.ObservabilityPlugin.Companion.LOG_PREFIX
+import org.opensearch.observability.settings.PluginSettings
+import org.opensearch.observability.util.SecureIndexClient
+import org.opensearch.observability.util.logger
+
+object IntegrationIndex {
+    private val log by logger(IntegrationIndex::class.java)
+    private const val INDEX_NAME = ".opensearch-integrations"
+    private const val INTEGRATIONS_MAPPING_FILE_NAME = "observability-mapping.yml"
```
[the remark this answers is not in the document]
Comment (YANG-DB) [reply src/main/kotlin/org/opensearch/integrations/index/IntegrationIndex.kt:23]: same here

File: src/main/kotlin/org/opensearch/integrations/index/IntegrationIndex.kt [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+    }
+
+    /**
+     * Create observability object
+     *
+     * @param observabilityObjectDoc
+     * @param id
+     * @return object id if successful, otherwise null
+     */
+    fun createIntegrationObject(integrationObjectDoc: Integration, id: String? = null): String? {
```
Comment (YANG-DB) [inline src/main/kotlin/org/opensearch/integrations/index/IntegrationIndex.kt:113]: we also need a delete integration object

File: src/main/kotlin/org/opensearch/integrations/model/Integration.kt [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+import org.opensearch.common.io.stream.StreamInput
+import org.opensearch.common.io.stream.StreamOutput
+import org.opensearch.common.xcontent.XContentParserUtils
+import org.opensearch.core.xcontent.ToXContent
+import org.opensearch.core.xcontent.XContentBuilder
+import org.opensearch.core.xcontent.XContentParser
+import org.opensearch.observability.ObservabilityPlugin.Companion.LOG_PREFIX
+import org.opensearch.observability.util.logger
+
+data class Integration(
```
Comment (YANG-DB) [inline src/main/kotlin/org/opensearch/integrations/model/Integration.kt:12]: this should be auto created from our Integration.schema file

File: src/main/resources/integrations-mapping.yml
Comment (YANG-DB) [inline src/main/resources/integrations-mapping.yml:1]: use [this schema](https://github.com/opensearch-project/observability/blob/main/src/main/resources/schema/system/integration.schema) as the base of the mapping file and keep all the mapping in that location

File: src/main/resources/integrations-settings.yml
Comment (YANG-DB) [inline src/main/resources/integrations-settings.yml:1]: move this file to the resources /schema/ system folder

**Review by YANG-DB — CHANGES_REQUESTED** [review]

File: src/main/kotlin/org/opensearch/integrations/model/Integration.kt (hunk 3 above)
Comment (Swiddis) [reply src/main/kotlin/org/opensearch/integrations/model/Integration.kt:12]: Out of scope for this PR, I think

File: src/main/kotlin/org/opensearch/integrations/index/IntegrationIndex.kt (hunk 2 above)
Comment (Swiddis) [reply src/main/kotlin/org/opensearch/integrations/index/IntegrationIndex.kt:113]: Out of scope for this PR

**Review by derek-ho — APPROVED** [review]

**Pull request merged**